posthog/hogql/database/database.py

```diff
@@ -143,6 +142,26 @@ def add_warehouse_tables(self, **field_definitions: Any):
             self._warehouse_table_names.append(f_name)
 
 
+def _use_person_properties_from_events(database: Database) -> None:
+    database.events.fields["person"] = FieldTraverser(chain=["poe"])
+
+
+def _use_person_id_from_person_overrides(database: Database) -> None:
+    database.events.fields["event_person_id"] = StringDatabaseField(name="person_id")
+    database.events.fields["override"] = LazyJoin(
+        from_field=["event_person_id"],
+        join_table=PersonOverridesTable(),
+        join_function=join_with_person_overrides_table,
+    )
+    database.events.fields["person_id"] = ExpressionField(
+        name="person_id",
+        expr=parse_expr(
+            "ifNull(nullIf(override.override_person_id, '00000000-0000-0000-0000-000000000000'), event_person_id)",
+            start=None,
+        ),
+    )
```

Comment:
Note that this is still using the old "v2" overrides table (which is not fully backfilled but is being kept up-to-date with realtime data) as that is the only one right now that is backwards compatible with legacy queries.

I'll follow up with a separate change that allows opting into the new "v3" table which _is_ fully backfilled (and is also kept up-to-date with realtime data), but nobody should be opted into that path that isn't either _completely_ on HogQL (for _all_ queries), _or_ okay with some queries/insights using different sets of overrides that have different data as HogQL will use v3, legacy will be stuck on v2 (how big of a discrepancy this is depends on the specific team and how often they merge/split persons.)

Reply:
WIP of new/v3/`person_distinct_id_overrides` support for HogQL _only_ is here: https://github.com/PostHog/posthog/compare/hogql-persons-on-events...hogql-persons-on-events-v3-overrides